### ISSUE TITLE
HystrixIntegrationTest fails with camel-2.19.x

### ISSUE THREAD

**Issue by tdiesler** [wildfly-extras/wildfly-camel#1721]

**Comment by jamesnetherton** [comment]
This is related to a change in camel-undertow and the way in which the UndertowClient is constructed. 

The code is now (using a specific ClassLoader):
```
UndertowClient.getInstance(camelContext.getApplicationContextClassLoader());
```

Where as before it was (using UndertowClient ClassLoader):
```
UndertowClient.getInstance();
```

This causes problems for WFC because we do not export undertow paths & and services onto the deployment class path.

Thus, the camel-undertow producer is currently broken for us in 2.19.

**Comment by jamesnetherton** [comment]
Today I learned..... When you deploy a WAR, WildFly automatically exports some of the Undertow packages.

For a WAR deployment, this works OOTB:

`UndertowClient instance = UndertowClient.getInstance();`

For a JAR deployment you get a `ClassNotFoundException`.

This probably explains why the test suite never picked up the issue with the Undertow producer, because the test case happens to deploy using a `WebArchive`.

**Comment by jamesnetherton** [comment]
Fixed on the camel-2.19.x branch.
